node/pkg/adminrpc/adminserver.go

```diff
@@ -1162,7 +1162,18 @@ func (s *nodePrivilegedService) SignExistingVAA(ctx context.Context, req *nodev1
 	}
 
 	// Add local signature
-	newVAA.AddSignature(s.gk, uint8(localGuardianIndex))
+	sig, err := s.guardianSigner.Sign(v.SigningDigest().Bytes())
+	if err != nil {
+		panic(err)
```

Comment:
Any chance we can do better than `panic` here? I see that the old method with `AddSignature` also panicked if the signature failed but maybe there's another way to handle errors here.

Reply:
I think for now this panic is okay. I think the original thinking was that signing is just a calculation, so if it fails,  something is really wrong. I think when we start adding signers that rely on external resources, we will need to rethink this. Having an external failure cause the guardian to roll in panics doesn't sound like a good plan.